Hibernate EnumType needs additional TypeHint

I am using spring-data-jpa with Hibernate. An Entity attribute annotated with @Enumerated results in a ClassNotFoundException.
Adding an additional TypeHint solves this, but I think it should be added to HibernateJpaHints.

kind regards

```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'entityManagerFactory' defined in class path resource [org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaConfiguration.class]: Invocation of init method
failed; nested exception is org.hibernate.boot.registry.classloading.spi.ClassLoadingException: Unable to load class [org.hibernate.type.EnumType]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1786) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:602) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:524) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[na:na]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[na:na]
        at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1154) ~[na:na]
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:908) ~[na:na]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583) ~[na:na]
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:144) ~[na:na]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:769) ~[com.example.oshiftboot.OshiftBootApplication:na]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:761) ~[com.example.oshiftboot.OshiftBootApplication:na]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:426) ~[com.example.oshiftboot.OshiftBootApplication:na]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:326) ~[com.example.oshiftboot.OshiftBootApplication:na]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1313) ~[com.example.oshiftboot.OshiftBootApplication:na]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1302) ~[com.example.oshiftboot.OshiftBootApplication:na]
        at com.example.oshiftboot.OshiftBootApplication.main(OshiftBootApplication.java:40) ~[com.example.oshiftboot.OshiftBootApplication:na]
Caused by: org.hibernate.boot.registry.classloading.spi.ClassLoadingException: Unable to load class [org.hibernate.type.EnumType]
        at org.hibernate.boot.registry.classloading.internal.ClassLoaderServiceImpl.classForName(ClassLoaderServiceImpl.java:133) ~[na:na]
        at org.hibernate.boot.internal.ClassLoaderAccessImpl.classForName(ClassLoaderAccessImpl.java:57) ~[na:na]
        at org.hibernate.cfg.annotations.SimpleValueBinder.fillSimpleValue(SimpleValueBinder.java:536) ~[na:na]
        at org.hibernate.cfg.SetSimpleValueTypeSecondPass.doSecondPass(SetSimpleValueTypeSecondPass.java:25) ~[na:na]
        at org.hibernate.boot.internal.InFlightMetadataCollectorImpl.processSecondPasses(InFlightMetadataCollectorImpl.java:1693) ~[na:na]
        at org.hibernate.boot.internal.InFlightMetadataCollectorImpl.processSecondPasses(InFlightMetadataCollectorImpl.java:1651) ~[na:na]
        at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.complete(MetadataBuildingProcess.java:295) ~[na:na]
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.metadata(EntityManagerFactoryBuilderImpl.java:1224) ~[na:na]
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:1255) ~[na:na]
        at org.springframework.orm.jpa.vendor.SpringHibernateJpaPersistenceProvider.createContainerEntityManagerFactory(SpringHibernateJpaPersistenceProvider.java:58) ~[na:na]
        at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.createNativeEntityManagerFactory(LocalContainerEntityManagerFactoryBean.java:365) ~[com.example.oshiftboot.OshiftBootApplication:5.3.5]
        at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.buildNativeEntityManagerFactory(AbstractEntityManagerFactoryBean.java:409) ~[com.example.oshiftboot.OshiftBootApplication:5.3.5]
        at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.afterPropertiesSet(AbstractEntityManagerFactoryBean.java:396) ~[com.example.oshiftboot.OshiftBootApplication:5.3.5]
        at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.afterPropertiesSet(LocalContainerEntityManagerFactoryBean.java:341) ~[com.example.oshiftboot.OshiftBootApplication:5.3.5]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1845) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1782) ~[na:na]
        ... 17 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.hibernate.type.EnumType
        at com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:60) ~[na:na]
        at java.lang.Class.forName(DynamicHub.java:1260) ~[na:na]
        at org.hibernate.boot.registry.classloading.internal.ClassLoaderServiceImpl.classForName(ClassLoaderServiceImpl.java:130) ~[na:na]
        ... 32 common frames omitted
```

--- a/samples/data-jpa/src/main/java/app/main/model/Foo.java
+++ b/samples/data-jpa/src/main/java/app/main/model/Foo.java
@@ -1,6 +1,8 @@
 package app.main.model;
+
 import javax.persistence.CascadeType;
 import javax.persistence.Entity;
+import javax.persistence.Enumerated;
 import javax.persistence.GeneratedValue;
 import javax.persistence.GenerationType;
 import javax.persistence.Id;
@@ -9,29 +11,32 @@ import javax.persistence.OneToOne;
 @Entity
 public class Foo {
 
-    @Id
-    @GeneratedValue(strategy=GenerationType.AUTO)
-    private long id;
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private long id;
+
 	private String value;
+
+	@Enumerated
+	private SomeEnum enumerated;
+
 	@OneToOne(cascade = CascadeType.ALL)
 	private Flurb flurb;
+
 	public Foo() {
 	}
+
 	public Foo(String value) {
 		this.value = value;
 	}
+
 	public String getValue() {
 		return this.value;
 	}
+
 	public void setValue(String value) {
 		this.value = value;
 	}
-    @Override
-    public String toString() {
-        return String.format(
-                "Foo[id=%d, value='%s']",
-                id, value);
-    }
 
 	public Flurb getFlurb() {
 		return flurb;
@@ -40,4 +45,16 @@ public class Foo {
 	public void setFlurb(Flurb flurb) {
 		this.flurb = flurb;
 	}
+
+	enum SomeEnum {
+		ONE, TWO
+	}
+
+	@Override
+	public String toString() {
+		return String.format(
+				"Foo[id=%d, value='%s']",
+				id, value);
+	}
+
 }

--- a/spring-native-configuration/src/main/java/org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaHints.java
+++ b/spring-native-configuration/src/main/java/org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaHints.java
@@ -51,6 +51,7 @@ import org.hibernate.tuple.component.PojoComponentTuplizer;
 import org.hibernate.tuple.entity.AbstractEntityTuplizer;
 import org.hibernate.tuple.entity.EntityTuplizer;
 import org.hibernate.tuple.entity.PojoEntityTuplizer;
+import org.hibernate.type.EnumType;
 import org.springframework.boot.orm.jpa.hibernate.SpringImplicitNamingStrategy;
 import org.springframework.boot.orm.jpa.hibernate.SpringPhysicalNamingStrategy;
 import org.springframework.data.jpa.repository.support.SimpleJpaRepository;
@@ -82,9 +83,9 @@ import java.util.EventListener;
 						HqlTokenTypes.class, SqlTokenTypes.class, GeneratedOrderByFragmentRendererTokenTypes.class,
 				}, access = AccessBits.DECLARED_CONSTRUCTORS),
 				@TypeHint(types = {
-						// petclinic
 						Repository.class,
-						PersistenceContext.class, MappedSuperclass.class, Column.class, ManyToOne.class, OneToMany.class, OneToOne.class, JoinColumn.class, Table.class, Transient.class
+						PersistenceContext.class, MappedSuperclass.class, Column.class, ManyToOne.class, OneToMany.class,
+						OneToOne.class, JoinColumn.class, Table.class, Transient.class
 				}, access = AccessBits.CLASS | AccessBits.DECLARED_METHODS),
 				@TypeHint(types = {
 						SessionImpl.class,
@@ -117,7 +118,9 @@ import java.util.EventListener;
 						NoJtaPlatform.class,
 						//org.hibernate.service.jta.platform.internal.NoJtaPlatform.class
 						EntityManagerMessageLogger.class, // CoreMessageLogger.class,
-						Session.class, EventListener.class
+						Session.class, EventListener.class,
+
+						EnumType.class
 				}, typeNames = {
 						"org.hibernate.internal.EntityManagerMessageLogger_$logger",
 						"org.hibernate.service.jta.platform.internal.NoJtaPlatform",
@@ -134,7 +137,6 @@ import java.util.EventListener;
 						OneToManyPersister.class, JoinedSubclassEntityPersister.class, SingleTableEntityPersister.class, UnionSubclassEntityPersister.class,
 						BasicCollectionPersister.class,
 				}, access = AccessBits.DECLARED_CONSTRUCTORS),
-
 				// AST nodes from the org.hibernate.hql.internal.ast.tree package
 				// These are necessary when HQL/JPQL queries are used.
 				@TypeHint(types = {

